addons/sale/models/sale_order.py

```diff
@@ -611,6 +627,56 @@ def _compute_amount_undiscounted(self):
                 total += (line.price_subtotal * 100)/(100-line.discount) if line.discount != 100 else (line.price_unit * line.product_uom_qty)
             order.amount_undiscounted = total
 
+    @api.depends('client_order_ref', 'date_order', 'origin', 'partner_id', 'tax_totals')
+    def _compute_duplicated_order_ids(self):
+        order_to_duplicate_orders = self._fetch_duplicate_orders()
+        for order in self:
+            order.duplicated_order_ids = [Command.set(order_to_duplicate_orders.get(order.id, []))]
+
+    def _fetch_duplicate_orders(self):
+        """ Fectch duplicated orders.
+
+        :return: Dictionary mapping order to it's related duplicated orders.
+        :rtype: dict
+        """
+        orders = self.filtered(lambda order: order.id and order.client_order_ref)
+        if not orders:
+            return {}
+
+        used_fields = (
+            'company_id',
+            'partner_id',
+            'client_order_ref',
+            'origin',
+            'date_order',
+            'state',
+            'amount_total',
+        )
+        self.env['sale.order'].flush_model(used_fields)
+
+        result = self.env.execute_query(SQL("""
+            SELECT
+                sale_order.id AS order_id,
+                array_agg(duplicate_order.id) AS duplicate_ids
+              FROM sale_order
+              JOIN sale_order AS duplicate_order
+                ON sale_order.company_id = duplicate_order.company_id
+                 AND sale_order.id != duplicate_order.id
+                 AND duplicate_order.state != 'cancel'
+                 AND sale_order.partner_id = duplicate_order.partner_id
+                 AND sale_order.amount_total = duplicate_order.amount_total
```

Comment:
I don't think we should enforce the amounts to be equal.
What if I import a purchase order, then the client adds other products to the PO, I import the new version of the XML file, and I've two orders with the same customer reference, but different amounts (and I should ideally delete or cancel the first one).

Reply:
Done remove amount dependency
Also update purchase export little bit to have `client_order_ref` unique per customer depending on PO name instead partner_ref :see_no_evil: